src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -33,8 +31,16 @@ private static Task FlushAsync(WritableBuffer writeBuffer)
             var awaitable = writeBuffer.FlushAsync();
             if (awaitable.IsCompleted)
             {
-                awaitable.GetResult();
-                return _completedTask;
+                try
+                {
+                    awaitable.GetResult();
+                }
+                catch(Exception e)
```

Comment:
p.s. if there was a fully declared type instead of `var` then I wouldn't have this confusion. Just saying... 😝